truss/remote/baseten/core.py

```diff
@@ -103,15 +125,22 @@ def create_chain(
                 raise ValueError(
                     f'Environment "{environment}" does not exist. You can create environments in the Chains UI.'
                 ) from e
+
             raise e
+    elif environment and environment != PRODUCTION_ENVIRONMENT_NAME:
```

Comment:
This check should be before l.114 or not?

Reply:
I think logically it should be equivalent to the pre-existing code. Is this a bug?

Reply:
It might be a bug. There is no usage of `environment` after this `elif` section - so what's the point of validating it then?

Reply:
I think it's because the `else` case that follows needs to assume that the user is intending to push to production.